src/openapi.ts

```diff
@@ -0,0 +1,119 @@
+import { sortObjectKeys, isRefObject, isPlainObject } from "./utils";
+import { AsyncAPIDocument, ConvertOpenAPIFunction, ConvertOptions, OpenAPIDocument } from "./interfaces";
+
+export const converters: Record<string, ConvertOpenAPIFunction > = {
+  'openapi': from_openapi_to_asyncapi,
+}
+
+function from_openapi_to_asyncapi(openapi: OpenAPIDocument, options: ConvertOptions): AsyncAPIDocument{
+    convertName(openapi)
+
+    convertInfoObject(openapi);
+    if (isPlainObject(openapi.servers)) {
+      openapi.servers = convertServerObjects(openapi.servers, openapi);
+    }
+
+    return sortObjectKeys(
+        openapi,
+        ['asyncapi', 'info', 'defaultContentType', 'servers', 'channels', 'operations', 'components']
+    )
+}
+
+function convertName(openapi: OpenAPIDocument): AsyncAPIDocument {
+    let { openapi: version } = openapi;
+    openapi.asyncapi = version;
```

Comment:
Since we are converting from one document type (OpenAPI) to another,  would it not make sense to create an empty AsyncAPI object and apply it to that instead of changing the openapi object?